### ISSUE TITLE
systemd-networkd-wait-online service wait for IPv6 network discovery after static IPv4 is configured

### ISSUE THREAD

**Issue by paulli2017** [systemd/systemd#27779]
### systemd version the issue has been seen with

252

### Used distribution

custome/yocto

### Linux kernel version used

5.10.154.10

### CPU architectures issue was seen on

aarch64

### Component

systemd-networkd-wait-online

### Expected behaviour you didn't see

The network configuration is static IPv4 support, and the systemd-networkd-wait-online service is expected to report link is configured and network is online as long as the static IPv4 address is configured, skipping IPv6 network discovery configuration, which delayed wait-online service time. We want network online to be faster.

### Unexpected behaviour you saw

From test, the systemd-networkd-wait-online service took 2 seconds, which includes more than 1.2 seconds on IPv6 network discovery configuration. From the systemd-netword source code, it seems that systemd-networkd-wait-online service needs to 
wait both static IPv4 address and IPv6 network discovery to be configured before it announces the link is configured and network
is online. The wait-online service is thus delayed by the IPv6 network discovery configuration.

### Steps to reproduce the problem

Here is the network configuration to check the issue:

[Network]
DHCP=No
Address=<PREFIX>.138.15/25
Gateway=<PREFIX>.138.1
DNS=<PREFIX>.99.167...

[IPv6AcceptRA]
DHCPv6Client=no
     

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
...
[    1.171503] (t-on[429]: systemd-networkd-wait-online.service: Executing: /usr/lib/systemd/systemd-networkd-wait-online
[    1.174346] systemd-networkd-wait-online[429]: Found link 1
[    1.174871] systemd-networkd-wait-online[429]: lo: link is ignored
...
[    1.603098] systemd-networkd[391]: enP8p1s0f0np0: link_check_ready(): static addresses are not configured.
...
[    1.608904] systemd-networkd[391]: enP8p1s0f0np0: Addresses set
...
[    1.609824] systemd-networkd[391]: enP8p1s0f0np0: Routes set
...
[    3.070317] systemd-networkd[391]: enP8p1s0f0np0: NDISC: Sent Router Solicitation, next solicitation in 3s
[    3.072865] systemd-networkd[391]: enP8p1s0f0np0: NDISC: Received Router Advertisement: flags MANAGED preference medium lifetime 1800 sec
...
[    3.074386] systemd-networkd[391]: enP8p1s0f0np0: SLAAC addresses and routes set.
[    3.074465] systemd-networkd[391]: enP8p1s0f0np0: link_check_ready(): IPv4LL:no DHCPv4:no DHCPv6:no DHCP-PD:no NDisc:yes
...
[    3.074582] systemd-networkd[391]: enP8p1s0f0np0: State changed: configuring -> configured
[    3.075692] systemd-networkd-wait-online[429]: enP8p1s0f0np0: link is configured by networkd and online.
...
```

**Comment by tomhughes** [comment]
Does setting `RequiredFamilyForOnline=ipv4` not work? The default is `any` which means it will wait for both.

**Comment by yuwata** [comment]
Fix is waiting in #27826. If possible, could you test the PR?

**Comment by glima** [comment]
> Fix is waiting in #27826. If possible, could you test the PR?

@paulli2017 did you see that? Would you please help backport and test?

**Comment by paulli2017** [comment]
> Fix is waiting in #27826. If possible, could you test the PR?

@yuwata I ported this changelist to my local system and tried to test it on my local system. I can see the wait-online service time improves a lot. It used to be about 2 second, now it is about 400 ms. It works well. thanks

**Comment by paulli2017** [comment]
> > Fix is waiting in #27826. If possible, could you test the PR?
> 
> @paulli2017 did you see that? Would you please help backport and test?

@glima Yeah, I tested the change and it works well. thanks
